src/geometry/euclidean/sphere/basic.lean

```diff
@@ -29,18 +29,21 @@ open_locale real_inner_product_space
 
 namespace euclidean_geometry
 
-variables {V : Type*} (P : Type*) [inner_product_space ℝ V] [metric_space P] [normed_add_torsor V P]
+variables {V : Type*} (P : Type*)
 
 open finite_dimensional
 
 /-- A `sphere P` bundles a `center` and `radius`. This definition does not require the radius to
 be positive; that should be given as a hypothesis to lemmas that require it. -/
-@[ext] structure sphere :=
+@[ext] structure sphere [metric_space P] :=
```

Comment:
Note that this isn't a change to the definition of `sphere`; the resulting type is unchanged

Reply:
It is. Now `sphere` takes argument `[metric_space P]` and doesn't use it. Is it intentional? If yes, then I find this comment misleading. Why do we want this change?
bors r-

Reply:
Otherwise LGTM.
bors d+

Reply:
> It is. Now `sphere` takes argument `[metric_space P]` and doesn't use it.

`sphere` had that argument [before this PR too](https://leanprover-community.github.io/mathlib_docs/geometry/euclidean/sphere/basic.html#euclidean_geometry.sphere)

> Is it intentional?

I am intentionally not changing it in this PR because my goal was not to refactor the whole file.

> If yes, then I find this comment misleading. Why do we want this change?

There is no change here

> And why not, e.g., `pseudo_metric_space`? bors r-

I agree that would be reasonable, but I'd rather leave further generalizations to the larger sort of refactor that @jsm28 describes.

Reply:
bors merge